lib/error/error.h

```diff
@@ -99,6 +99,19 @@ const char* error_severity_get_name(uint8_t severity);
         }                                                           \
     } while (0)
 
+/** Call this macro to only log one out of N occurence. This is useful in tight call loops. */
+#define WARNING_N(n, text, ...)                                     \
```

Comment:
no templates? :slightly_smiling_face: `WARNING<N>` with a default to 1 would be nicer

Reply:
```suggestion
template<int n = 1> void WARNING(const std::string& text, ...) {
        static int _cnt = 0;                                       
        if ((_cnt++ % n) == 0 && g_error_fct.warning) {             
            struct error e = error_generate(ERROR_SEVERITY_WARNING, 
                                            (text),                 
                                            (__FILE__),             
                                            __LINE__);              
            g_error_fct.warning(&e, ##__VA_ARGS__);                 
        }                                                           
}
```

Reply:
This would not work for several reasons:

1. This API is used by pure C callers. While we discuss from time to time migrating to complete C++, this is not something that happened yet.
2. We need preprocessor expansion for __LINE__ and __FILE__, not sure how that would play with templates.
3. The API is not very clean: What does `WARNING<4>("foobar")` mean?

Also, wouldn't that lead to a much bigger code size ?

Reply:
`WARNING_N` what does it mean? both are arbitrary IMO but points 1-2 remain valid same for code size

Reply:
Okay, what I meant is that templates here do not provide additional cleanliness compared to what can be achieved in good old  C.

Reply:
Renamed to `WARNING_EVERY_N` to try to make it more readable.